vulture_os/toolkit/api_parser/trendmicro_visionone/trendmicro_visionone.py

```diff
@@ -0,0 +1,180 @@
+        headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token,
+                   'Accept': 'application/json',
+                   'TMV1-Filter': ''}
+
+        r = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
+
+        if r.status_code != 200:
+            raise TrendmicroVisionOneAPIError(
+                f"Error on URL: {self.URL_BASE + url_path} Status: {r.status_code} Reason/Content: {r.content}")
+
+        return r.json()['items']
+
+    def _get_OAT(self, since=None, to=None):
+
+        url_path = '/v3.0/audit/logs'
+
+        query_params = {'detectedStartDateTime': since,
+                        'detectedEndDateTime': to,
+                        'ingestedStartDateTime': since,
+                        'ingestedEndDateTime': to,
+                        'top': 50}
+        headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token,
+                   'Accept': 'application/json',
+                   'TMV1-Filter': ''}
+
+        r = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
+
+        if r.status_code != 200:
+            raise TrendmicroVisionOneAPIError(
+                f"Error on URL: {self.URL_BASE + url_path} Status: {r.status_code} Reason/Content: {r.content}")
+
+        return r.json()['items']
+
+    def _format_OAT_log(self, log):
+        return log["detail"]
+
+
+    def execute(self):
+
+        for kind in ["alerts", "oat", "audit"]:
+
+            try:
+                since = getattr(self, f"trendmicro_visionone_{kind}_timestamp") or (datetime.now() - timedelta(days=2))
+                to = min(timezone.now(), since + timedelta(hours=24))
+
+                msg = f"Parser gets {kind} logs from {since} to {to}"
+                logger.info(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
+                start_time = since.strftime("%Y-%m-%dT%H:%M:%SZ")
+                end_time = to.strftime("%Y-%m-%dT%H:%M:%SZ")
+                if kind == "alerts":
+                    logs = self._get_alerts(start_time, end_time)
+                    self.write_to_file(logs)
+                elif kind == "audit":
+                    logs = self._get_auditlogs(start_time, end_time)
+                    self.write_to_file(logs)
+                elif kind == "oat":
+                    logs = self._get_OAT(start_time, end_time)
+                    self.write_to_file([self._format_OAT_log(log) for log in logs])
+
+                self.update_lock()
+
+            except Exception as e:
+                msg = f"Failed on {kind} logs, between time of {since} and {to} : {e} "
+                logger.error(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
+                logger.exception(f"[{__parser__}]:execute: {e}", extra={'frontend': str(self.frontend)})
+
+            setattr(self.frontend, f"trendmicro_visionone_{kind}_timestamp", to, tz=timezone.utc)
```

Comment:
could hava avoided using 3 different timestamps for this technology if you fixed the `to` before beginning recovering logs for each kind !